site/docs/skylark/backward-compatibility.md

```diff
@@ -662,5 +663,29 @@ my_rule = rule(
 *   Default: `false`
 *   Introduced in: `0.18.0`
 
+### Disallow `cfg = "data"`
+
+`cfg = "data"` is a no-op that incorrectly gives the impression dependencies under
+it are built in some special "data" mode:
+
+```python
+my_rule = rule(
+    ...
+    "some_attr": attr.label_list(
+        cfg = "data"  # This line does nothing
+    )
+)
+```
+
+Because this is non-functional and confusing, it's being removed outright
+([#6153](https://github.com/bazelbuild/bazel/issues/6153)). The functionality this
+syntax implies will be provided by
+[Starlark build configuration](https://github.com/bazelbuild/bazel/issues/5574).
+
```

Comment:
Say explicitly that with this flag enabled, using cfg = "data" will be an error.

Reply:
Done.